pkg/githubreconciler/issuemanager/session.go

```diff
@@ -38,50 +47,73 @@ func (s *IssueSession[T]) hasSkipLabel(issue *github.Issue) bool {
 	return false
 }
 
-// UpsertMany creates or updates multiple issues based on the provided data.
-// It uses the Equal method on T to determine which existing issues correspond to which desired data.
+// Reconcile reconciles the issue state with the desired state by creating, updating, and closing issues.
+// It performs a complete reconciliation:
+// - Creates new issues for desired states without matching existing issues
+// - Updates existing issues that match desired states (if content changed)
+// - Closes existing issues that don't match any desired state
+// Issues with the skip label are preserved and not modified in any phase.
 // The pathLabel is automatically added to the provided labels.
 // Returns a slice of issue URLs in the same order as the input data.
-func (s *IssueSession[T]) UpsertMany(
+func (s *IssueSession[T]) Reconcile(
 	ctx context.Context,
 	desired []*T,
 	labels []string,
+	closeMessage string,
 ) ([]string, error) {
 	log := clog.FromContext(ctx)
 
+	// Check for duplicate desired states
+	for i := 0; i < len(desired); i++ {
+		for j := i + 1; j < len(desired); j++ {
+			if (*desired[i]).Equal(*desired[j]) {
+				return nil, fmt.Errorf("duplicate desired state detected: entries at index %d and %d have matching identity fields", i, j)
```

Comment:
"have matching identity fields" seems overly specific to a particular Equal impl?